### PR TITLE
Convert PreconditionChebyshev::set_initial_guess_kernel to Kokkos

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: include/deal.II/lac/precondition.h [hunk 1]
```diff
@@ -3416,16 +3379,19 @@ namespace internal
...
-        1 + (n_local_elements - 1) / CUDAWrappers::block_size;
-      set_initial_guess_kernel<<<n_blocks, CUDAWrappers::block_size>>>(
-        first_local_range, n_local_elements, vector.get_values());
-      AssertCudaKernel();
-
+      Number *   values_ptr       = vector.get_values();
+      Kokkos::RangePolicy<typename MemorySpace::kokkos_space::execution_space,
+                          Kokkos::IndexType<types::global_dof_index>>
+        policy(0, n_local_elements);
+      Kokkos::parallel_for(
```
Comment (peterrum) [inline include/deal.II/lac/precondition.h:3386]: I hope this won't be the new standard how to write code in deal.II...

**Review by peterrum — APPROVED** [review]
Looks good to me.

File: include/deal.II/lac/precondition.h (hunk 1 above)
Comment (masterleinad) [reply include/deal.II/lac/precondition.h:3386]: Would you mind elaborating?

File: include/deal.II/lac/precondition.h [hunk 2]
```diff
@@ -3416,16 +3379,19 @@ namespace internal
...
+        "PreconditionChebyshev::set_initial_guess",
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply include/deal.II/lac/precondition.h:3387]: Well, this helps `Kokkos`' tools for profiling.

[the remark this answers is not in the document]
Comment (bangerth) [reply include/deal.II/lac/precondition.h:3387]: One wished that they had made that the last argument to the function, with a default to `""`. But having it is a good idea.

**Review by kronbichler — APPROVED** [review]
Looks good to me. I also think that passing strings is much uglier than having the right C++ function name to start with, but lambdas are what they are. On the performance side, these are big enough loops and not in the innermost level of algorithms. Hence I see no problem (my insistence is because the string is more than 22 characters, so it will allocate memory I assume, and thus have a minimal run time of ~3e-7 seconds).

**Comment by bangerth** [comment]
Is it worth just standardizing on passing `__PRETTY_FUNCTION__` everywhere we call these kernels?

**Comment by Rombur** [comment]
> Is it worth just standardizing on passing __PRETTY_FUNCTION__ everywhere we call these kernels?

We tried something similar in a different project but the problem is that the names are too long. You can't read them easily in your profiler. Like Daniel said the name is optional but it's very useful.

**Comment by kronbichler** [comment]
OK, let's go with this solution, which is a good progress.

**Pull request merged**